api/src/pcapi/core/offers/schemas.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any
+
+from pydantic.v1 import EmailStr
+from pydantic.v1 import HttpUrl
+from pydantic.v1 import validator
+
+from pcapi.core.offers import models as offers_models
+from pcapi.routes.serialization import BaseModel
+from pcapi.serialization.utils import to_camel
+from pcapi.validation.routes.offers import check_offer_name_length_is_valid
+
+
+class PostDraftOfferBodyModel(BaseModel):
+    name: str
+    subcategory_id: str
+    venue_id: int
+    description: str | None = None
+    url: HttpUrl | None = None
+
+    @validator("name", pre=True)
+    def validate_name(cls, name: str, values: dict) -> str:
+        check_offer_name_length_is_valid(name)
+        return name
+
+    class Config:
+        alias_generator = to_camel
+        extra = "forbid"
+
+
+class PatchDraftOfferBodyModel(BaseModel):
+    name: str | None = None
+    description: str | None = None
+    url: HttpUrl | None = None
+
+    @validator("name", pre=True)
+    def validate_name(cls, name: str, values: dict) -> str:
+        check_offer_name_length_is_valid(name)
+        return name
+
+    class Config:
+        alias_generator = to_camel
+        extra = "forbid"
+
+
+class PatchDraftOfferDetailsBodyModel(BaseModel):
+    audio_disability_compliant: bool | None = None
+    mental_disability_compliant: bool | None = None
+    motor_disability_compliant: bool | None = None
+    visual_disability_compliant: bool | None = None
+    booking_contact: EmailStr | None = None
+    booking_email: EmailStr | None = None
+    duration_minutes: int | None = None
+    external_ticket_office_url: HttpUrl | None = None
+    extra_data: Any = None
+    is_duo: bool | None = None
+    is_national: bool | None = None
+    withdrawal_delay: int | None = None
+    withdrawal_details: str | None = None
+    withdrawal_type: offers_models.WithdrawalTypeEnum | None = None
```

Comment:
```suggestion
    url: HttpUrl | None = None
    withdrawal_delay: int | None = None
    withdrawal_details: str | None = None
    withdrawal_type: offers_models.WithdrawalTypeEnum | None = None

    # is_national must be placed after url so that the validator
    # can access the url field in the dict of values
    # (which contains only previously validated fields)
    is_national: bool | None = None
```